spacy_llm/tasks/entity_linker/task.py

```diff
@@ -0,0 +1,240 @@
+
+        parse_responses (TaskResponseParser[Self]): Callable for parsing LLM responses for this task.
+        prompt_example_type (Type[FewshotExample[Self]]): Type to use for fewshot examples.
+        prompt_examples (Optional[List[FewshotExample[Self]]]): Optional list of few-shot examples to include in prompts.
+        template (str): Prompt template passed to the model.
+        scorer (Scorer): Scorer function.
+        """
+        super().__init__(
+            parse_responses=parse_responses,
+            prompt_example_type=prompt_example_type,
+            template=template,
+            prompt_examples=prompt_examples,
+        )
+        self._scorer = scorer
+        self._candidate_selector: Optional[CandidateSelector] = None
+
+        # Exclude mentions without candidates from prompt, if set.
+        self._auto_nil = True
+        # Store, per doc and entity, whether candidates could be found.
+        self._has_ent_cands: List[List[bool]] = []
+
+    def initialize(
+        self,
+        get_examples: Callable[[], Iterable["Example"]],
+        nlp: Language,
+        candidate_selector: CandidateSelector,
+        n_prompt_examples: int = 0,
+    ) -> None:
+        """Initialize entity linking task.
+        get_examples (Callable[[], Iterable["Example"]]): Callable that provides examples
+            for initialization.
+        nlp (Language): Language instance.
+        candidate_selector (CandidateSelector): Factory for a candidate selection callable
+            returning candidates for a given Span and context.
+        n_prompt_examples (int): How many prompt examples to infer from the provided Example objects.
+            0 by default. Takes all examples if set to -1.
+
+        """
+        super()._initialize(
+            get_examples=get_examples,
+            nlp=nlp,
+            n_prompt_examples=n_prompt_examples,
+            fetch_entity_info=self.fetch_entity_info,
+        )
+        self._candidate_selector = candidate_selector
+        if isinstance(self._candidate_selector, InitializableCandidateSelector):
+            self._candidate_selector.initialize(nlp.vocab)
+
+    def generate_prompts(self, docs: Iterable[Doc]) -> Iterable[str]:
+        environment = jinja2.Environment()
+        _template = environment.from_string(self._template)
+        # Reset auto-nil attributes for new batch of docs.
+        self._has_ent_cands = []
+
+        for i_doc, doc in enumerate(docs):
+            cands_ents, _ = self.fetch_entity_info(doc)
+            # Determine which ents have candidates and should be included in prompt.
+            has_cands = [
+                {cand_ent.id for cand_ent in cand_ents} != {EntityLinker.NIL}
+                or not self._auto_nil
+                for cand_ents in cands_ents
+            ]
+            self._has_ent_cands.append(has_cands)
+
+            # To improve: if a doc has no entities (with candidates), skip prompt altogether?
```

Comment:
That would be good, yes

Reply:
I recommend doing this in a follow-up PR, as this would this would modify core behavior (1:1 relationship between doc and prompt) and should be properly tested.